src/ngx_http_lua_uri.c

```diff
@@ -55,6 +56,8 @@ ngx_http_lua_ngx_req_set_uri(lua_State *L)
         return luaL_error(L, "attempt to use zero-length uri");
     }
 
+    len = ngx_http_lua_safe_uri_len(p, len);
```

Comment:
Ditto. Let's check the return value if there is unsafe bytes, we should throw out a Lua exception.